meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/util/ZkMetaClientUtil.java

```diff
@@ -262,6 +262,14 @@ public static MetaClientInterface.ConnectState translateKeeperStateToMetaClientC
     }
   }
 
+  public static MetaClientInterface.Stat convertZkStatToStat(
+      org.apache.zookeeper.data.Stat zkStat) {
+    return new MetaClientInterface.Stat(
+        convertZkEntryModeToMetaClientEntryMode(zkStat.getEphemeralOwner()), zkStat.getVersion(),
+        zkStat.getCtime(), zkStat.getMtime(),
+        EphemeralType.TTL.getValue(zkStat.getEphemeralOwner()));
```

Comment:
I remember there was a conversation a while ago where `zkStat.getEphemeralOwner() `is buggy where it doesn't actually return the proper owner (if I recall correctly we contacted the open source committee for apache ZooKeeper). Not sure if this was fixed?

Reply:
Yes you are right, this API doesn't actually return the proper owner.
We still need to code based on public document. :D